File: php/features.php
<?php
declare(strict_types=1);

// Letscount SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class LetscountFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new LetscountBaseFeature();
            case "test":
                return new LetscountTestFeature();
            default:
                return new LetscountBaseFeature();
        }
    }
}

File: py/letscount_sdk/features.py
# Letscount SDK feature factory

from letscount_sdk.feature.base_feature import LetscountBaseFeature
from letscount_sdk.feature.test_feature import LetscountTestFeature


def _make_feature(name):
    features = {
        "base": lambda: LetscountBaseFeature(),
        "test": lambda: LetscountTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
